R/helper-functions.R

```diff
@@ -418,12 +422,34 @@ hashPassword <- function(password) {
 #'   add_data(x = t2)$
 #'   add_sparklines(sparklines = sparklines)
 #'
+#' # create sparkline groups
+#' sparklines <- c(
+#'   create_sparklines("Sheet 2", "A2:L6;", "M2:M6", markers = "1"),
+#'   create_sparklines(
+#'     "Sheet 2", "A7:L7;A9:L9", "M7;M9", type = "stacked", negative = "1"
+#'   ),
+#'   create_sparklines(
+#'     "Sheet 2", "A8:L8;A10:L13", "M8;M10:M13",
+#'     type = "column", first = "1"
+#'    ),
+#'   create_sparklines(
+#'     "Sheet 2", "A2:L13", "A14:L14", type = "column", first = "1",
+#'     direction = "col"
+#'   )
+#' )
+#'
+#' wb <- wb$
+#'   add_worksheet("Sheet 2")$
+#'   add_data(x = t2)$
+#'   add_sparklines(sparklines = sparklines)
+#'
 #' @export
 create_sparklines <- function(
     sheet                  = current_sheet(),
     dims,
     sqref,
     type                   = NULL,
+    direction              = "row",
```

Comment:
Please move this to the end (in front of the `...`). I like to minimize potentially breaking changes like changing the argument order

Reply:
Also shouldn't this be `NULL` by default. Otherwise this PR will break e.g.  https://github.com/JanMarvin/openxlsx2/discussions/921

Reply:
1. Fixed. Moved the `direction` argument to the end. Concerning the second issue. 

2. Concerning the second issue. I checked the code in #921 and it works fine with the PR. The reason is that in my implementation the `direction=` argument will only have an effect it the `dims=` spans both multiple columns and rows, i.e. `split_dims` will not split the cell range spanning only one column or row (different to your proposal). 

I already was aware that this is not optimal. But based on the review I now realized that this also results in a bug in the knife-edge case where one wants to create a group of sparklines for each cell of one row or one column, e.g. `create_sparklines("Sheet 1", "A2:K2", "A34:K34", direction = "col")` will result in an error. I will have a look and propose a fix.